bin/check-aggregate-output.rb

```diff
@@ -0,0 +1,209 @@
+#!/usr/bin/env ruby
+#
+# Check Aggregate Output
+# ===
+# This check is very similar to check-aggregate.rb, it checks the Sensu aggregate
+# api and returns Critical / Warning / OK based on how many aggregate results
+# match the options you configure.
+#
+# Authors
+# ===
+# AJ Bourg, @ajbourg
+# Sean Porter, @portertech
+#
+# Copyright 2017
+#
+# Released under the same terms as Sensu (the MIT license); see
+# LICENSE for details.
+
+require 'sensu-plugin/check/cli'
+require 'rest-client'
+require 'json'
+
+class CheckAggregateOutput < Sensu::Plugin::Check::CLI
+  option :api,
+         short: '-a URL',
+         long: '--api URL',
+         description: 'Sensu API URL',
+         default: if ENV['SENSU_API']
+                    ENV['SENSU_API'] + ':4567'
+                  elsif ENV['SENSU_API_URL']
+                    ENV['SENSU_API_URL']
+                  else
+                    'http://localhost:4567'
+                  end
+
+  option :insecure,
+         short: '-k',
+         boolean: true,
+         description: 'Enabling insecure connections',
+         default: false
+
+  option :user,
+         short: '-u USER',
+         long: '--user USER',
+         description: 'Sensu API USER'
+
+  option :password,
+         short: '-p PASSWORD',
+         long: '--password PASSWORD',
+         description: 'Sensu API PASSWORD'
+
+  option :timeout,
+         short: '-t SECONDS',
+         long: '--timeout SECONDS',
+         description: 'Sensu API connection timeout in SECONDS',
+         proc: proc(&:to_i),
+         default: 30
+
+  option :check,
+         short: '-c CHECK',
+         long: '--check CHECK',
+         description: 'Aggregate CHECK name',
+         required: true
+
+  option :age,
+         short: '-A SECONDS',
+         long: '--age SECONDS',
+         description: 'Minimum aggregate age in SECONDS, time since check request issued',
+         default: 30,
+         proc: proc(&:to_i)
+
+  option :message,
+         short: '-M MESSAGE',
+         long: '--message MESSAGE',
+         description: 'A custom error MESSAGE'
+
+  option :collect_output,
+         short: '-o',
+         long: '--output',
+         boolean: true,
+         description: 'Collects all non-ok outputs',
+         default: true
+
+  option :warning,
+         short: '-W PERCENT',
+         long: '--warning PERCENT',
+         description: 'PERCENT warning before warning (can be change with --ignore-severity)',
+         proc: proc(&:to_i)
```

Comment:
I think these should be floats rather than integers if they are percentages.

For example:
```ruby
$ irb
irb(main):001:0> 2.5.class
=> Float
irb(main):002:0> 2.5*100
=> 250.0
irb(main):003:0> 2.5.to_i*100
=> 200
irb(main):04:0> 99.99.to_f*100
=> 9999.0
irb(main):005:0> 99.99.to_i*100
=> 9900
```

This increases the number of aggregated results to alert on by a small but non trivial amount. While it's not off by much in many cases it could be more important if say you need four or five nines and are using this for SLA reporting. I am gonna look at the other check as I bet it is present there as well. If so I will file an issue and create a pull request to it.